File: bhs-project/src/services/productService.js
import axios from "axios"

export default class ProductService{
    getProducts(){
        return axios.get("http://localhost:8080/api/Products/getAll")
    }

    getByProductName(productName){
        return axios.get("http://localhost:8080/api/Products/getByProductName?productName="+ productName)
    }
}

File: Mekanik/src/main/java/bhs/Mekanik/business/abstracts/ProductService.java
package bhs.Mekanik.business.abstracts;

import java.util.List;

import bhs.Mekanik.core.utilities.result.DataResult;
import bhs.Mekanik.core.utilities.result.Result;
import bhs.Mekanik.entities.concretes.Product;

public interface ProductService {
	DataResult <List<Product>> getAll();
	Result add(Product product);
	DataResult<Product> getByProductName (String productName);
}
